src/main/java/com/aws/greengrass/shadowmanager/sync/model/LocalUpdateSyncRequest.java

```diff
@@ -201,25 +188,19 @@ public void execute(SyncContext context) throws SkipSyncRequestException, Confli
      */
     @Override
     public boolean isUpdateNecessary(SyncContext context) throws SkipSyncRequestException, UnknownShadowException {
-        // synchronizing access to shadow sync information because it is accessed by multiple threads.
-        // In this method, shadow sync information is accessed within context of the Nucleus'
-        // MqttClient event loop thread.
-        // The other place is the 'execute' method of this class, which runs on the sync thread.
-        synchronized (context.getSynchronizeHelper().getThingShadowLock(this)) {
-            //TODO: store this information in a return object to avoid unnecessary calls to DAO.
-            ShadowDocument shadowDocument;
-            try {
-                shadowDocument = new ShadowDocument(updateDocument);
-            } catch (IOException | InvalidRequestParametersException e) {
-                throw new SkipSyncRequestException(e);
-            }
+        //TODO: store this information in a return object to avoid unnecessary calls to DAO.
```

Comment:
removed the `synchronized` block since not we're not reading the sync info from multiple threads anymore